symphony/bdk/core/service/stream/stream_service.py

```diff
@@ -0,0 +1,263 @@
+        :param room_id: the id of the room in which to add the user.
+        :return: None
+        """
+        await self._room_membership_api.v1_room_id_membership_add_post(
+            payload=UserId(id=user_id), id=room_id,
+            session_token=await self._auth_session.session_token)
+
+    async def remove_member_from_room(self, user_id: int, room_id: int):
+        """Removes a member from an existing room.
+        Wraps the `Remove Member <https://developers.symphony.com/restapi/reference#remove-member>`_ endpoint.
+
+        :param user_id: the id of the user to be removed from the room.
+        :param room_id: the id of the room from which to remove the user.
+        :return: None
+        """
+        await self._room_membership_api.v1_room_id_membership_remove_post(
+            payload=UserId(id=user_id), id=room_id,
+            session_token=await self._auth_session.session_token)
+
+    async def share(self, stream_id: str, content: ShareContent) -> V2Message:
+        """Share third-party content, such as a news article, into the specified stream.
+        The stream can be a chat room, an IM, or an MIM.
+        Wraps the `Share <https://developers.symphony.com/restapi/reference#share-v3>`_ endpoint.
+
+        :param stream_id: the id of the stream in which to share the given content.
+        :param content: the third-party content to be shared.
+        :return: the created message.
+        """
+        return await self._share_api.v3_stream_sid_share_post(
+            sid=stream_id, share_content=content, session_token=await self._auth_session.session_token,
+            key_manager_token=await self._auth_session.key_manager_token)
+
+    async def promote_user_to_room_owner(self, user_id: int, room_id: str):
+        """Promotes user to owner of the chat room.
+        Wraps the `Promote Owner <https://developers.symphony.com/restapi/reference#promote-owner>`_ endpoint.
+
+        :param user_id: the id of the user to be promoted as owner.
+        :param room_id: the room id.
+        :return: None
+        """
+        await self._room_membership_api.v1_room_id_membership_promote_owner_post(
+            id=room_id, payload=UserId(id=user_id), session_token=await self._auth_session.session_token)
+
+    async def demote_owner_to_room_participant(self, user_id: int, room_id: str):
+        """Demotes room owner to a participant in the chat room.
+        Wraps the `Demote Owner <https://developers.symphony.com/restapi/reference#demote-owner>`_ endpoint.
+
+        :param user_id: the id of the room owner to be demoted.
+        :param room_id: the room id.
+        :return: None
+        """
+        await self._room_membership_api.v1_room_id_membership_demote_owner_post(
+            id=room_id, payload=UserId(id=user_id), session_token=await self._auth_session.session_token)
+
+    async def create_im_or_mim(self, user_ids: [int]) -> Stream:
```

Comment:
in the java one we just called it create i think
and we also have create_room below

i'm not sure how you do proper method overloading in python, if you tend to name methods differently or just put all the possible parameters and document them

Reply:
There is no such thing as method overloading in python. So either we have two different method names (as I did) or we put one function with all params and discriminate based on which param is not None (which I don't like)

Reply:
I would prefer the first option to make them explicit 🤔